pyccel/parser/syntactic.py

```diff
@@ -372,18 +372,22 @@ def _visit_AugAssign(self, stmt):
         return expr
 
     def _visit_arguments(self, stmt):
-        arguments = []
+
         if stmt.vararg or stmt.kwarg:
             errors.report(VARARGS, symbol = stmt,
                     severity='fatal')
 
+        arguments       = []
         if stmt.args:
             n_expl = len(stmt.args)-len(stmt.defaults)
-            arguments += [Argument(a.arg) for a in stmt.args[:n_expl]]
-            arguments += [ValuedArgument(Argument(a.arg),self._visit(d)) for a,d in zip(stmt.args[n_expl:],stmt.defaults)]
+            positional_args        = [Argument(a.arg, annotation=self._visit(a.annotation)) for a in stmt.args[:n_expl]]
+            valued_arguments       = [ValuedArgument(Argument(a.arg, annotation=self._visit(a.annotation)),\
+                                      self._visit(d)) for a,d in zip(stmt.args[n_expl:],stmt.defaults)]
+            arguments              = positional_args + valued_arguments
         if stmt.kwonlyargs:
-            arguments += [ValuedArgument(Argument(a.arg),self._visit(d), kwonly=True) if d is not None
-                        else Argument(a.arg, kwonly=True) for a,d in zip(stmt.kwonlyargs,stmt.kw_defaults)]
+            kwonly_arguments       = [ValuedArgument(Argument(a.arg, annotation=self._visit(a.annotation)),self._visit(d), kwonly=True) if d is not None
```

Comment:
This line barely fits on screen. Could you break it up a bit more? Or maybe there should be a function for this?

Reply:
yes you are right I can break it into two lines